modules/xfem/include/base/XFEM.h

```diff
@@ -424,26 +440,19 @@ class XFEM : public XFEMInterface
    * @param elem2       The second child element
    */
   void storeMaterialPropertiesForElements(const Elem * parent_elem,
-                                          const std::vector<const Elem *> & elems,
-                                          const GeometricCutUserObject * gcuo);
+                                          const std::vector<const Elem *> & elems);
 
   /**
    * Helper function to store the material properties of a healed element
-   * @param parent_elem The parent element
-   * @param cut_elem    The element being cut
-   * @param elem_from   The element to copy material properties from
+   * @param elem       The cut element to restore material properties to.
+   * @param elem_from  The element to copy material properties from.
    */
-  void setMaterialPropertiesForElement(const Elem * parent_elem,
-                                       const Elem * cut_elem,
-                                       const GeometricCutUserObject * gcuo);
+  void setMaterialPropertiesForElement(const Elem * elem, const Elem * elem_from);
 
   /**
-   * Determine which cut subdomain of the element belongs to relative to the cut
-   * @param parent_elem The parent element
-   * @param cut_elem    The element being cut
-   * @param gcuo        The GeometricCutUserObject for the cut
+   * Randomly pick one node on the element that is in the physical domain
```

Comment:
I'd reword this as:

Return the first node in the provided element that is found to be in the physical domain

...because it's not actually random.